src/emulator/apphosting/config.spec.ts

```diff
@@ -9,76 +8,125 @@ import * as configImport from "../../apphosting/config";
+
+      const apphostingConfig = await getLocalAppHostingConfiguration("./");
+
+      expect(JSON.stringify(apphostingConfig.environmentVariables)).to.equal(
+        JSON.stringify([
+          { variable: "randomEnvOne", value: "envOne" },
+          { variable: "randomEnvTwo", value: "envTwo" },
+          { variable: "randomEnvThree", value: "envThree" },
+        ]),
+      );
+
+      expect(JSON.stringify(apphostingConfig.secrets)).to.equal(
+        JSON.stringify([
+          { variable: "randomSecretOne", secret: "secretOne" },
+          { variable: "randomSecretTwo", secret: "secretTwo" },
+          { variable: "randomSecretThree", secret: "secretThree" },
+        ]),
+      );
+    });
+
     it("should combine apphosting yaml files according to precedence", async () => {
-      pathExistsStub.returns(true);
+      listAppHostingFilesInPathStub.returns([
+        "/parent/cwd/apphosting.yaml",
+        "/parent/apphosting.local.yaml",
+      ]);
 
       // Second config takes precedence
-      const apphostingYamlConfigTwo = AppHostingYamlConfig.empty();
-      const apphostingYamlConfigThree = AppHostingYamlConfig.empty();
-
-      apphostingYamlConfigTwo.addEnvironmentVariable({
-        variable: "randomEnvOne",
-        value: "envOne",
-      });
-      apphostingYamlConfigTwo.addEnvironmentVariable({
-        variable: "randomEnvTwo",
-        value: "envTwo",
-      });
-      apphostingYamlConfigTwo.addEnvironmentVariable({
-        variable: "randomEnvThree",
-        value: "envThree",
-      });
-
-      apphostingYamlConfigTwo.addSecret({ variable: "randomSecretOne", secret: "secretOne" });
-      apphostingYamlConfigTwo.addSecret({ variable: "randomSecretTwo", secret: "secretTwo" });
-      apphostingYamlConfigTwo.addSecret({ variable: "randomSecretThree", secret: "secretThree" });
-
-      apphostingYamlConfigThree.addEnvironmentVariable({
-        variable: "randomEnvOne",
-        value: "envOne",
-      });
-      apphostingYamlConfigThree.addEnvironmentVariable({
-        variable: "randomEnvTwo",
-        value: "blah",
-      });
-      apphostingYamlConfigThree.addEnvironmentVariable({
-        variable: "randomEnvFour",
-        value: "envFour",
-      });
-
-      apphostingYamlConfigThree.addSecret({ variable: "randomSecretOne", secret: "bleh" });
-      apphostingYamlConfigThree.addSecret({ variable: "randomSecretTwo", secret: "secretTwo" });
-      apphostingYamlConfigThree.addSecret({ variable: "randomSecretFour", secret: "secretFour" });
-
-      loadAppHostingYamlStub.onFirstCall().returns(apphostingYamlConfigThree);
-      loadAppHostingYamlStub.onSecondCall().returns(apphostingYamlConfigTwo);
-
-      const apphostingConfig = await getLocalAppHostingConfiguration("test");
+      loadAppHostingYamlStub.onFirstCall().returns(apphostingYamlConfigTwo);
+      loadAppHostingYamlStub.onSecondCall().returns(apphostingYamlConfigOne);
```

Comment:
nit* pretty tricky since it's unclear which file is associated with these reads.

When I was writing tests for similar functionality, I ended up creating a tmpdir and actually wrote files. Not asking you to do it here but something to consider if you find later that these test are brittle:

https://github.com/firebase/firebase-tools/blob/7a500afbcdbbd631751c7c8a606c50aeb18930ac/src/functions/env.spec.ts#L292

Reply:
ya this is a good way to go about it, will take a look!